download.go

```diff
@@ -0,0 +1,123 @@
+package ytdlp
+
+import (
+	"encoding/json"
+	"strconv"
+	"strings"
+)
+
+// DownloadProgressFunc is a callback function that is called whenever there is a progress update.
+type DownloadProgressFunc = func(DownloadProgress)
+
+// DownloadStatus represents the status of a download.
+type DownloadStatus string
+
+const (
+	// StatusPreProcessing represents the download status when the download is pre-processing.
+	StatusStarting DownloadStatus = "starting"
+	// StatusDownloading represents the download status when the download is in progress.
+	StatusDownloading DownloadStatus = "downloading"
+	// StatusPostProcessing represents the download status when the download is post-processing.
+	StatusPostProcessing DownloadStatus = "post_processing"
+	// StatusFinished represents the download status when the download is finished.
+	StatusFinished DownloadStatus = "finished"
+)
+
+// DownloadProgress represents the progress of a download.
+type DownloadProgress struct {
+	ID            string
```

Comment:
Can we add documentation to these fields? E.g. where they are populated from, what `Total` means, etc. If possible, include the unit of measurement in the field name (e.g. is `Total` the `TotalBytes`?)